### PR TITLE
Include assembly location and app domain bin path in search for sqlite3.dll

### LINKED ISSUE #278: sqlite3.dll not found in ASP.NET 4 applications
Repro

Create ASP.NET 4 web application
Add Microsoft.Data.Sqlite 1.0.0
Add code that calls `SqliteConnection.Open()`

Error
DllImportException "sqlite3.dll" not found

Reason
Despite the build appropriately places sqlite3.dll in to bin/x86 and bin/x64, ASP.NET sets `AppDomain.CurrentDomain.BaseDirectory` to the project root, but NativeLibraryLoader needs to use "$projectRoot/bin"

### PR DESCRIPTION
**Pull request by natemcmaster**
Fix #278 

cc @bricelam

### CODE REVIEW

**Comment by natemcmaster** [comment]
cc @davidfowl

File: src/Microsoft.Data.Sqlite/Utilities/NativeLibraryLoader.cs [hunk 1]
```diff
@@ -37,29 +39,50 @@ public static bool TryLoad(string dllName)
...
-                return false;
+                yield return assemblyDir;
+                yield return Path.Combine(assemblyDir, GetArchitecture());
             }
 
-            var handle = LoadLibraryEx(dllPath, IntPtr.Zero, LOAD_WITH_ALTERED_SEARCH_PATH);
-            Marshal.ThrowExceptionForHR(Marshal.GetLastWin32Error());
-
-            return handle != IntPtr.Zero;
+            var privateBinPath = AppDomain.CurrentDomain.SetupInformation?.PrivateBinPath;
```
Comment (davidfowl) [inline src/Microsoft.Data.Sqlite/Utilities/NativeLibraryLoader.cs]: I'd look here first. When is SetupInformation null?

Comment (natemcmaster) [inline src/Microsoft.Data.Sqlite/Utilities/NativeLibraryLoader.cs]: SetupInformation is never null AFAIK.  Updated.

File: src/Microsoft.Data.Sqlite/Utilities/NativeLibraryLoader.cs [hunk 2]
```diff
@@ -37,29 +39,53 @@ public static bool TryLoad(string dllName)
...
+        private static IEnumerable<string> GetSearchPaths()
+        {
+            yield return AppDomain.CurrentDomain.BaseDirectory;
+            yield return Path.Combine(AppDomain.CurrentDomain.BaseDirectory, GetArchitecture());
+
+            var privateBinPath = AppDomain.CurrentDomain.SetupInformation.PrivateBinPath;
+            if (!string.IsNullOrEmpty(privateBinPath))
             {
-                return false;
+                foreach (var path in privateBinPath.Split(new [] { ';' }, StringSplitOptions.RemoveEmptyEntries))
```
Comment (bricelam) [inline src/Microsoft.Data.Sqlite/Utilities/NativeLibraryLoader.cs]: `Path.DirectorySeparatorChar`?

**Comment by bricelam** [comment]
:shipit:

File: src/Microsoft.Data.Sqlite/Utilities/NativeLibraryLoader.cs (hunk 2 above)
Comment (natemcmaster) [inline src/Microsoft.Data.Sqlite/Utilities/NativeLibraryLoader.cs]: Assuming you meant `Path.PathSeparator`?

Comment (bricelam) [inline src/Microsoft.Data.Sqlite/Utilities/NativeLibraryLoader.cs]: That's the one :smile:

**Pull request merged**